plugins/jenkins-backend/src/service/router.ts

```diff
@@ -20,22 +20,38 @@ import Router from 'express-promise-router';
 import { Logger } from 'winston';
 import { JenkinsInfoProvider } from './jenkinsInfoProvider';
 import { JenkinsApiImpl } from './jenkinsApi';
-import { PermissionEvaluator } from '@backstage/plugin-permission-common';
+import {
+  PermissionAuthorizer,
+  PermissionEvaluator,
+  toPermissionEvaluator,
+} from '@backstage/plugin-permission-common';
 import { getBearerTokenFromAuthorizationHeader } from '@backstage/plugin-auth-node';
 import { stringifyEntityRef } from '@backstage/catalog-model';
 
 export interface RouterOptions {
   logger: Logger;
   jenkinsInfoProvider: JenkinsInfoProvider;
-  permissions?: PermissionEvaluator;
+  permissions?: PermissionEvaluator | PermissionAuthorizer;
 }
 
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { jenkinsInfoProvider } = options;
-
-  const jenkinsApi = new JenkinsApiImpl(options.permissions);
+  const { jenkinsInfoProvider, permissions, logger } = options;
+
+  let permissionEvaluator: PermissionEvaluator | undefined;
+  if (permissions?.hasOwnProperty('query')) {
```

Comment:
I think you wanna check this with `in` instead right? Most of the time this is gonna be on the prototype. Maybe to keep the code slim we can structure things so that all you need to do is 

```ts
permissionEvaluator = toPermissionEvaluator(permissions)
```

Reply:
yeah the thing is I also would like to show a warning without checking things twice and without passing the logger to `toPermissionEvaluator` 😅